docs/tasks/tls/managing-tls-in-a-cluster.md

```diff
@@ -0,0 +1,189 @@
+stashed in the `.spec.request` field. We are also requesting a certificate
+with the "digital signature", "key encipherment", and "server auth" key
+usages. We support all key usages and extended key usages listed
+[here](https://godoc.org/k8s.io/client-go/pkg/apis/certificates/v1beta1#KeyUsage)
+so you can request client certificates and other certificates using this same
+API.
+
+The CSR should now be visible from the API
+in a Pending state. You can see it by running:
+
+```console
+$ kubectl describe csr my-svc.my-namespace
+Name:                   my-svc.my-namespace
+Labels:                 <none>
+Annotations:            <none>
+CreationTimestamp:      Tue, 21 Mar 2017 07:03:51 -0700
+Requesting User:        yourname@example.com
+Status:                 Pending
+Subject:
+        Common Name:    my-svc.my-namespace.svc.cluster.local
+        Serial Number:
+Subject Alternative Names:
+        DNS Names:      my-svc.my-namespace.svc.cluster.local
+        IP Addresses:   172.168.0.24
+                        10.0.34.2
+Events: <none>
+```
+
+### Step 3. Get the Certificate Signing Request Approved
+
+Approving the certificate signing request is either done by an automated
+approval process or on a one off basis by a cluster administrator. More
+information on what this involves below.
+
+### Step 4. Download the Certificate and Use It
+
+Once the CSR is signed and approved you should see the following:
+
+```console
+$ kubectl get csr
+NAME                  AGE       REQUESTOR               CONDITION
+my-svc.my-namespace   10m       yourname@example.com    Approved,Issued
+```
+
+You can download the issued certificate and save it to a `server.crt` file by
+running the following:
+
+```console
+$ kubectl get csr my-svc.my-namespace -o jsonpath='{.status.certificate}' \
+    | base64 -d > server.crt
+```
+
+Now you can use `server.crt` and `server-key.pem` as the keypair to start your
+HTTPS server.
+
+## Approving Certificate Signing Requests
+
+A Kubernetes administrator (with appropriate permissions) can manually approve
+(or deny) Certificate Signing Requests by using the `kubectl certificate
+approve` and `kubectl certificate deny` commands. However if you intend
+to make heavy usage of this API, you might consider writing an automated
+certificates controller.
+
+Whether a machine or a human, the role of the approver is to verify that
+the CSR satisfies two requirements:
+
+1. The subject of the CSR controls the private key used to sign the CSR. This
+   addresses the threat of a third party masquerading as an authorized subject.
+2. The subject of the CSR is authorized to act in the requested context. This
+   addresses the threat of an undesired subject joining the cluster.
+
+If and only if these two requirements are met, the approver should approve
+the CSR and otherwise should deny the CSR.
+
+
+## A Word of **Warning** on the Approval Permission
+
+The ability to approve CSRs decides who trusts who within the cluster. This
```

Comment:
Also a bigger discussion than just this review, but I'd like to discuss how we could go about partitioning off certs that are core kubernetes from certs created by users for various purposes. It seems like we just have one flat structure where users could accidentally issue powerful certs for system pieces they weren't expecting to, or be tricked into doing so.